src/main/java/Entities/Board.java

```diff
@@ -37,8 +46,7 @@ public List<Player> getPlayers(){
      * @param player The player that must be removed.
      */
     public void removePlayer(Player player){
-        List<Player> newPlayers = players;
-        newPlayers.remove(player);
```

Comment:
Good catch on this. The previous version I think had the idea of being correct but it forgot to pass newPlayers into this.players. But either way this implementation is a lot better as doing it the previous way may be a waste of resources due to the introduction of newPlayers.